credentials/fetcher_default.go

```diff
@@ -124,19 +128,21 @@ func (s *FetcherDefault) fetchParallel(ctx context.Context, locations []url.URL)
 	select {
 	case <-ctx.Done():
 		s.l.Errorf("Ignoring JSON Web Keys from at least one URI because the request timed out waiting for a response.")
+		return fetchResult{true}
```

Comment:
Maybe return an error here instead? And then we check for errors in the caller code? That would be a bit more idiomatic go code :)

Reply:
Thanks for the honest feedback, I really appreciate comments like this - I am not that proficient with Go yet :)
Changed to `return error`, is now ok - or should I even improve the implementation with a custom error?